### ISSUE TITLE
[4.0] RFC Empty headings [a11y]

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#29909]
https://dequeuniversity.com/rules/axe/3.5/empty-heading

> Screen readers alert users to the presence of a heading tag. If the heading is empty or the text cannot be accessed, this could either confuse users or even prevent them from accessing information on the page's structure.

An example of where joomla _fails_ on this is the new user form as this line
`	<h2><?php echo $this->form->getValue('name'); ?></h2>
`
results in
`       <h2></h2>`

There are many ways to _resolve_ this
- 1. Check if its a new or existing record and only display the heading if its existing
```
<?php if ($this->item->id != 0): ?>
	<h2><?php echo $this->form->getValue('name'); ?></h2>
<?php endif; ?>
```
- 2. Display different text as the heading if its a new or existing record
```
<?php if ($this->item->id != 0): ?>
	<h2><?php echo Text::_('COM_USERS_VIEW_NEW_USER_TITLE'); ?></h2>
<?php else: ?>
	<h2><?php echo $this->form->getValue('name'); ?></h2>
<?php endif; ?>
```
3. and probably a lot lot more

**Comment by hans2103** [comment]
@brianteeman if I'm correct the public function for getValue of $this->form is declared in libraries/src/Form/Form.php

And this function has an `optional default value of the field value is empty.`

https://github.com/joomla/joomla-cms/blob/4e1013af2ff61e3407c7d2f17c777446d2332968/libraries/src/Form/Form.php#L539-L563

Agree?

let me propose possible solution numer 4. 
We can benefit the default value, instead of an if-then-else statement.

```
<h2><?php echo $this->form->getValue('name', null, Text::_('COM_USERS_VIEW_NEW_USER_TITLE')); ?></h2>
```

if no name is given, the default value (third parameter), will be used. 
What do you think of this solution?

**Comment by brianteeman** [comment]
Go for it. The only problem is that we will now have identical h1 and H2. Typing from my phone so can't be 100%

**Comment by hans2103** [comment]
@brianteeman Can you propose an alternative default value ?

**Comment by brianteeman** [comment]
I can't think of anything myself and I guess that's why it is currently empty? Maybe others can

**Comment by alikon** [comment]
as per #29920<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29909">issues.joomla.org/tracker/joomla-cms/29909</a>.</sub>

**Comment by brianteeman** [comment]
Please re-open - only one instance is fixed in that pr

**Comment by alikon** [comment]
right

**Comment by hans2103** [comment]
- [x] administrator/index.php?option=com_users&view=user&layout=edit
  https://github.com/joomla/joomla-cms/pull/29920
- [x] administrator/index.php?option=com_languages&view=language&layout=edit
  https://github.com/joomla/joomla-cms/pull/29939

**Comment by hans2103** [comment]
@brianteeman when more issues found, please inform me.

**Comment by brianteeman** [comment]
I had another look last night and I didnt see them - I was sure I found more than these two originally ;(

I will close it for now.
